odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -1387,6 +1387,59 @@ def _mocked_check_access_rights(records, operation, raise_exception=True):
             values = message.read(['attributes'])[0]['attributes'][0]
         self.assertEqual(values['value'], (tag.id, 'Test Tag'))
 
+    @users('test')
+    def test_no_access_error_when_fetching_property_field_default_values(self):
+        """We can read the child, but not the definition record.
+
+        Check that the user does not get an `AccessError` when creating a new
+        record having a property field whose property definition is stored on
+        a record the user does not have access to. The newly created record
+        should have the right schema and should be populated with the default
+        values stored on the property definition.
+        """
+
+        # Update the access rights of the parent model holding the property definition:
+        accesses = self.env['ir.model.access'].sudo().search([
+            ('model_id.model', '=', 'test_new_api.discussion')
+        ])
+        accesses.write({
+            'perm_read': False,
+            'perm_write': False,
+            'perm_create': False,
+            'perm_unlink': False
+        })
```

Comment:
Seems a bit hardcore, can't we mock "check_access_rights" for this model, similarly as done here:
https://github.com/odoo/odoo/blob/16.0/addons/test_mail/tests/test_mail_composer.py#L946